src/software/ai/hl/stp/play/defense/defense_play_fsm.cpp

```diff
@@ -2,24 +2,80 @@
 
 #include "software/ai/evaluation/defender_assignment.h"
 #include "software/ai/evaluation/enemy_threat.h"
+#include "software/logger/logger.h"
+
 
 DefensePlayFSM::DefensePlayFSM(TbotsProto::AiConfig ai_config)
     : DefensePlayFSMBase::DefensePlayFSMBase(ai_config)
 {
 }
 
-void DefensePlayFSM::defendAgainstThreats(const Update& event)
+bool DefensePlayFSM::shouldDefendAggressively(const Update& event)
+{
+    // If there is more attackers ahead of the ball than there
+    // is our own defenders we probably shouldn't press
+    // this can be replaced with smarter or stricter logic though
```

Comment:
you can make a new github issue for this and tag it here (but we probably won't prioritize it for now)